service/src/test/java/uk/gov/hmcts/reform/fpl/service/NoticeOfProceedingsServiceTest.java

```diff
@@ -1,66 +1,314 @@
+    private static final byte[] PDF = {1, 2, 3, 4, 5};
 
     @Autowired
     private NoticeOfProceedingsService noticeOfProceedingService;
 
+    @Autowired
+    private Time time;
+
+    @MockBean
+    private DocmosisDocumentGeneratorService docmosisDocumentGeneratorService;
+
+    @MockBean
+    private NoticeOfProceedingsTemplateDataGenerationService noticeOfProceedingsTemplateDataGenerationService;
+
+    @MockBean
+    private UploadDocumentService uploadDocumentService;
+
+    private LocalDateTime now;
+
+    @BeforeEach
+    void before() {
+        now = time.now();
+    }
+
+    @Test
+    void shouldSetHearingAndJudgeLabelWhenBothHearingAndJudgeInformationExistOnCaseData() {
+        CaseData caseData = CaseData.builder()
+            .allocatedJudge(buildAllocatedJudge())
+            .hearingDetails(createHearingBookings())
+            .build();
+
+        Map<String, Object> listAndLabel = noticeOfProceedingService.initNoticeOfProceeding(caseData);
+
+        assertThat(listAndLabel)
+            .extracting("proceedingLabel", "noticeOfProceedings")
+            .containsExactly(buildExpectedHearingLabel(), buildExpectedNoticeOfProceedingData());
+    }
+
+    @Test
+    void shouldSetAllocatedJudgeLabelWhenOnlyJudgeDataExistsOnCaseData() {
+        CaseData caseData = CaseData.builder()
+            .allocatedJudge(buildAllocatedJudge())
+            .build();
+
+        Map<String, Object> listAndLabel = noticeOfProceedingService.initNoticeOfProceeding(caseData);
+
+        assertThat(listAndLabel)
+            .extracting("proceedingLabel", "noticeOfProceedings")
+            .containsExactly(null, buildExpectedNoticeOfProceedingData());
+    }
+
+    @Test
+    void shouldSetHearingLabelWhenOnlyHearingDetailsExistOnCaseData() {
+        CaseData caseData = CaseData.builder()
+            .hearingDetails(createHearingBookings())
+            .build();
+
+        Map<String, Object> listAndLabel = noticeOfProceedingService.initNoticeOfProceeding(caseData);
+
+        assertThat(listAndLabel)
+            .extracting("proceedingLabel", "noticeOfProceedings")
+            .containsExactly(buildExpectedHearingLabel(), null);
+    }
+
+    @Test
+    void shouldSetNoticeOfProceedingJudgeToAllocatedJudgeWhenUseAllocatedJudgeHasBeenSelected() {
+        CaseData caseData = CaseData.builder()
+            .allocatedJudge(buildAllocatedJudge())
+            .noticeOfProceedings(NoticeOfProceedings.builder()
+                .judgeAndLegalAdvisor(JudgeAndLegalAdvisor.builder()
+                    .allocatedJudgeLabel("Allocated judge label")
+                    .useAllocatedJudge(YES.getValue())
+                    .build())
+                .build())
+            .build();
+
+        NoticeOfProceedings noticeOfProceedings = noticeOfProceedingService.setNoticeOfProceedingJudge(caseData);
+
+        assertThat(noticeOfProceedings.getJudgeAndLegalAdvisor().getJudgeTitle()).isEqualTo(HIS_HONOUR_JUDGE);
```

Comment:
might be better to build a test/expected JudgeAndLegalAdvisor and compare that way